Python/pystate.c

```diff
@@ -2195,15 +2195,12 @@ _PyInterpreterFrame *
 _PyThreadState_PushFrame(PyThreadState *tstate, size_t size)
 {
     assert(size < INT_MAX/sizeof(PyObject *));
-    PyObject **base = tstate->datastack_top;
-    PyObject **top = base + size;
-    if (top >= tstate->datastack_limit) {
-        base = push_chunk(tstate, (int)size);
+    if (_PyThreadState_HasStackSpace(tstate, (int)size)) {
+        _PyInterpreterFrame *res = (_PyInterpreterFrame *)tstate->datastack_top;
+        tstate->datastack_top += size;
```

Comment:
The old version did the addition for `PyObject **`.  Now we are doing the addition for `_PyInterpreterFrame *`.  Do they have the same size?

Reply:
We are still doing the addition for `PyObject **`.

Reply:
You are right!  I misread.